### PR TITLE
App Submission : Calibre-web

### PR DESCRIPTION
**Pull request by Xosten**
# App Submission

### App name
Calibre-web

### 256x256 SVG icon
https://svgur.com/s/j54

### Gallery images
Calibre-web-1
Calibre-web-2
Calibre-web-3

### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [ ] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [x]  [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

## Side notes
- Author: https://github.com/janeczku/calibre-web
- Docker container maintainer (shared by the author): https://github.com/linuxserver/docker-calibre-web
- In order to work, Calibre-Web needs a valid Calibre database. That's why an empty `metadata.db` file is located in `data/books/`
- Tested on an OCI ARM instance

### CODE REVIEW

File: calibre-web/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+version: "2.1"
+services:
+  app_proxy:
+    environment:
+      APP_HOST: $APP_CALIBRE_WEB_IP
+      APP_PORT: $APP_CALIBRE_WEB_PORT
+
+  calibre-web:
+    image: linuxserver/calibre-web:0.6.18@sha256:4ddccf9ca8cba38dfb6e46ba56828395102253a3ff78a449aa2bc812877d9d16
```
Comment (lukechilds) [inline calibre-web/docker-compose.yml:9]:
```suggestion
    image: linuxserver/calibre-web:0.6.18@sha256:f0e848b8c5a28c40eab4b5db88e02a78f8835fff6066b57921b57b2653fb7514
```

**Comment by lukechilds** [comment]
Thanks for this @Xosten, tested and working, although I'm a little confused about how to add books to the library.

Is it possible to use Calibre-web to upload ebook files and it will create the Calibre DB? Or is it assumed the Calibre DB is something that the user has created via some other means and this app just reads from it?

**Comment by Xosten** [comment]
> Thanks for this @Xosten, tested and working, although I'm a little confused about how to add books to the library.
> 
> Is it possible to use Calibre-web to upload ebook files and it will create the Calibre DB? Or is it assumed the Calibre DB is something that the user has created via some other means and this app just reads from it?

Thank you luke for reviewing the submission. I totally get what you mean, because I had the same in the beginning when I was trying to set up the container for the first time. 

Calibre-Web is a web-application version of Calibre Desktop. You can either import an existing database by replacing the empty one that I've added per default (it can't work without a DB file). If you want to upload straight to the web application, you need to enable the feature by going through `Settings > Edit Basic Configuration > Feature Configuration > Enable Upload`. The upload button will be shown at the top.

Then it's recommended to connect calibre to an email server, they suggest gmail as it's the most simple. This is for sending your ebooks straight to your Kindle ;). 

Does it make sense ? 

I will update the application description with the steps to enable the upload feature.

**Comment by lukechilds** [comment]
Screenshot 2022-08-01 at 17 15 45

Sweet, got it working! The first click was called `Admin` for me not `Settings` btw.

**Comment by Xosten** [comment]
Glad to hear it worked! Yes correct, it's called Settings for me as I'm using a theme. My bad 😅. Hopefully this app will be handy for others!

**Review by lukechilds — APPROVED** [review]

**Comment by lukechilds** [comment]
Tested and working great! @mayankchhabra this is ready for gallery assets 🚀

**Comment by mayankchhabra** [comment]
Really nice work packaging Calibre Web @Xosten! The app store gallery images and app icon are now live via [getumbrel/umbrel-apps-gallery](https://github.com/getumbrel/umbrel-apps-gallery/tree/master/calibre-web).

Good luck with the launch in the app store! 🚀

PS: If you'd like to update the gallery images or the app icon in the future, feel free to submit a PR to the above repo. Cheers!

**Comment by Xosten** [comment]
> Really nice work packaging Calibre Web @Xosten! The app store gallery images and app icon are now live via [getumbrel/umbrel-apps-gallery](https://github.com/getumbrel/umbrel-apps-gallery/tree/master/calibre-web).
> 
> Good luck with the launch in the app store! 🚀
> 
> PS: If you'd like to update the gallery images or the app icon in the future, feel free to submit a PR to the above repo. Cheers!

Thanks man! Let's bring as much handy apps to Umbrel!

**Pull request merged**